packages/slate-editor/src/modules/rich-formatting-menu/RichFormattingMenu.tsx

```diff
@@ -109,14 +109,24 @@ export function RichFormattingMenu({
     function handleLinkButtonClick() {
         if (!editor.selection) return;
 
-        Transforms.select(editor, editor.selection.anchor.path);
-
         const rangeRef = Editor.rangeRef(editor, editor.selection, {
             affinity: 'inward',
         });
 
         setLinkRange(rangeRef);
 
+        if (rangeRef.current) {
+            const hasLink = getCurrentLinkNode(editor, { at: rangeRef.current });
+            const selectedOnlyLink = Path.equals(
+                rangeRef.current.anchor.path,
+                rangeRef.current.focus.path,
+            );
```

Comment:
I think this should be called `isCollapsed`?

Reply:
I thought that `isCollapsed` means the same offset as well?